src/modules/Posts.ts

```diff
@@ -41,6 +41,12 @@ const posts = (state = initialState, action: Action<Post[]>) =>
 
 export const reducer = posts
 
+// ------------------------------------
+// Selectors
+// ------------------------------------
+
+export const postsSelector = state => state.posts || initialState
```

Comment:
For my understanding, we normally create selectors to grab specific pieces of state. So that when we refactor our state/logics, UI components are still fine.
Therefore, instead of having 1 selector to grab a whole `posts` state, I think it is better to have something like:
```javascript
  export const postsSelector = {
    getPosts: state => state.posts.data,
    getLoading: state => state.posts.loading,
    ...
  }
```

Reply:
It would be fine since typescript will catch error when change the name of our properties.